kernel/arch/dreamcast/hardware/maple/maple_init_shutdown.c

```diff
@@ -34,6 +35,55 @@
 
 */
 
+static void maple_dev_reset_cb(maple_state_t *st, maple_frame_t *frame) {
+    (void)st;
+
+    /* Unlock the frame */
+    maple_frame_unlock(frame);
+
+    /* Wake up! */
+    genwait_wake_all(frame);
+}
+
+void maple_dev_reset(maple_device_t *dev) {
+
+    assert(dev != NULL);
+
+    /* Lock the frame */
+    while(maple_frame_lock(&dev->frame) < 0)
+        thd_pass();
+
+    /* Reset the frame */
+    maple_frame_init(&dev->frame);
+    dev->frame.cmd = MAPLE_COMMAND_RESET;
+    dev->frame.dst_port = dev->port;
+    dev->frame.dst_unit = dev->unit;
+    dev->frame.length = 0;
+    dev->frame.callback = maple_dev_reset_cb;
+    maple_queue_frame(&dev->frame);
+
+    /* Wait for the device to accept it */
```

Comment:
Just as a note, tried to remove this, as we don't care what the response was after all there's no failure state other than not returning and there's nothing we'd really be able to do about it.

I know I'd discussed before the idea of  removing this or making it optional in 'fire-and-forget' commands (draw, beep, rumble) as it seems mostly useful for something where you need confirmation like the vmu reading and such. The problem is that the wait doesn't *just* ensure we get a return but that the command gets sent at all.

`maple_queue_frame`, as the name implies, just queues up the command to be sent. Normally there's little concern that it'll be sent in a timely fashion, but in this case we're doing it directly before shutting down all the maple hardware so in my testing it usually just failed to send the resets.

The right response though may be to have a test in `maple_hw_shutdown` ensure that the queue gets cleared before shutting down.